zerocopy-derive/src/lib.rs

```diff
@@ -510,105 +512,21 @@ const STRUCT_UNION_ALLOWED_REPR_COMBINATIONS: &[&[StructRepr]] = &[
 ];
 
 fn derive_try_from_bytes_enum(ast: &DeriveInput, enm: &DataEnum) -> proc_macro2::TokenStream {
-    if !enm.is_fieldless() {
-        return Error::new_spanned(ast, "only field-less enums can implement TryFromBytes")
-            .to_compile_error();
-    }
-
     let reprs = try_or_print!(ENUM_TRY_FROM_BYTES_CFG.validate_reprs(ast));
 
-    // Figure out whether the enum could in theory implement `FromBytes`.
-    let from_bytes = enum_size_from_repr(reprs.as_slice())
-        .map(|size| {
-            // As of this writing, `enm.is_fieldless()` is redundant since we've
-            // already checked for it and returned if the check failed. However, if
-            // we ever remove that check, then without a similar check here, this
-            // code would become unsound.
-            enm.is_fieldless() && enm.variants.len() == 1usize << size
-        })
-        .unwrap_or(false);
-
-    let variant_names = enm.variants.iter().map(|v| &v.ident);
-    let is_bit_valid_body = if from_bytes {
-        // If the enum could implement `FromBytes`, we can avoid emitting a
-        // match statement. This is faster to compile, and generates code which
-        // performs better.
-        quote!({
-            // Prevent an "unused" warning.
-            let _ = candidate;
-            // SAFETY: If the enum could implement `FromBytes`, then all bit
-            // patterns are valid. Thus, this is a sound implementation.
-            true
-        })
```

Comment:
Was it intentional to remove this optimization?

Reply:
Yes, we're going to replace this with #1303